lib/src/kbasesearchengine/parse/ObjectParser.java

```diff
@@ -18,12 +18,25 @@
 import us.kbase.common.service.UObject;
 
 public class ObjectParser {
-    
-    public static File prepareTempFile(File tempDir) throws IOException {
+    private static final String TEMP_FILE_PREFIX = "ws_srv_response_";
+    private static final String TEMP_FILE_SUFFIX = ".json";
+
+    /** Creates a temporary file in the specified directory. The file is deleted
+     * when the virtual machine terminates.
+     *
+     * @param tempDir
+     * @return temporaray file.
+     * @throws IOException If file cannot be created.
+     * @throws SecurityException If Security Manager denies access to delete the
+     * temporary file.
+     */
+    public static File prepareTempFile(File tempDir)
+            throws IOException, SecurityException {
         if (!tempDir.exists()) {
             tempDir.mkdirs();
         }
-        File tempFile = File.createTempFile("ws_srv_response_", ".json", tempDir);
+        File tempFile = File.createTempFile(TEMP_FILE_PREFIX, TEMP_FILE_SUFFIX, tempDir);
+        tempFile.deleteOnExit();
```

Comment:
I would prefer to handle this by clearing the temp dir on startup so that temp files can be inspected if there's some error. The temp files are deleted in a finally block in the indexer worker.